### ISSUE TITLE
[google_maps_flutter_ios] Add Swift Package Manager compatibility to google_maps_flutter_ios

### ISSUE THREAD

**Issue by vashworth** [flutter/flutter#146920]
Prerequisites:
- [ ] https://github.com/flutter/flutter/pull/146256
- [ ] https://github.com/flutter/flutter/issues/146901

---

Follow ["Converting an existing Objective-C Flutter Plugin to a Swift Package" instructions](https://github.com/flutter/flutter/pull/146256#issuecomment-2038565077) to make `google_maps_flutter_ios` plugin compatible with Swift Package Manager and CocoaPods.

**Comment by wangela** [comment]
The documentation is now at https://docs.flutter.dev/packages-and-plugins/swift-package-manager/for-plugin-authors#how-to-add-swift-package-manager-support-to-an-existing-flutter-plugin

**Comment by stuartmorgan** [comment]
Capturing some information and discussion from comments in issues and Discord:
- According to comments in https://github.com/flutter/packages/pull/8288, SPM support for `google_maps_ios` requires dropping support for iOS 14, which Flutter still supports.
- We could potentially support iOS 15+ with SPM, and tell people to disable SPM for this plugin if they want to continue supporting iOS 14. However:
  - It's not currently clear to me what error message someone would get if they had an existing app supporting iOS 14, updated to a version of Flutter that enabled SPM by default and updated their packages, and ended up with an app that could no longer support iOS 14.
  - It's not clear how someone would discover that they need to disable SPM to support iOS 14 if we don't do a breaking change to `google_maps_flutter` to make it extremely prominent in the CHANGELOG.

https://github.com/flutter/flutter/issues/94491#issuecomment-1896847437 summarizes a lot of discussion about the last time we had to do this (drop a version of iOS in a plugin while Flutter still supports that version of iOS). We try very hard not to do it, because the user experience is pretty bad when it's not a breaking change to the plugin (and if it is, that causes other user experience issues). Last time we added a new error message to `flutter` just to support doing it less disruptively, but I don't know that we have a comparable message implemented for SPM.

So the next steps here from my perspective would be:
- Capture (here) the exact error message someone would get if their build suddenly stopped supporting iOS 14 due to SPM dependency changes.
- Investigate whether we need to implement a better message than we currently have, as we did for CocoaPods build failures due to min version mismatch.

**Comment by stuartmorgan** [comment]
The comment above assumes that we do in fact need to drop iOS 14 for `google_maps_flutter` SPM support. @jokerttu, @wangela, could you elaborate as to why that is necessary?

I found https://github.com/googlemaps/ios-maps-sdk, which seems to be the source repo for the SPM package, and while [the 9.0 release](https://github.com/googlemaps/ios-maps-sdk/pull/23) requires iOS 15, as expected for that SDK version, there is a commit for an [8.4.0 release](https://github.com/googlemaps/ios-maps-sdk/pull/7) that [supports iOS 14](https://github.com/googlemaps/ios-maps-sdk/blob/4b24150b5a48a8624808b23f36a8e2804df333e4/Package.swift#L22). Why can't we include that version in our SPM dependency range, as we do with CocoaPods?

**Comment by jokerttu** [comment]
Thanks @stuartmorgan,

> The comment above assumes that we do in fact need to drop iOS 14 for `google_maps_flutter` SPM support. [@jokerttu](https://github.com/jokerttu), [@wangela](https://github.com/wangela), could you elaborate as to why that is necessary?
>
> (Is this related to [googlemaps/ios-maps-sdk#10 (comment)](https://github.com/googlemaps/ios-maps-sdk/issues/10#issuecomment-2096890239) ? There was never any clear resolution in that issue as to whether the 9.0 release did in fact fix the problem.)

My understanding was that there was a blocker with an earlier attempt to get SPM working, which was fixed in version 9.0.0 of the Google Maps SDK for iOS ([see discussion](https://github.com/googlemaps/ios-maps-sdk/issues/10#issuecomment-2332591495)).

I re-tested SPM support, including package ranges that support iOS 14, and I didn’t see any issues this time. Previously, I encountered compatibility problems, but they seem to have been related to my local setup or build environment. I’ll update the PR to include SPM support for iOS 14 as well.

Apologies for the extra inconvenience this caused. I'm glad if iOS 14 support won’t have to be dropped after all!

**Comment by loic-sharma** [comment]
> It's not currently clear to me what error message someone would get if they had an existing app supporting iOS 14, updated to a version of Flutter that enabled SPM by default and updated their packages, and ended up with an app that could no longer support iOS 14.

In this scenario, `flutter build ios` will result in the following error:

```
$ flutter build ios
...
Failed to build iOS app
Target Integrity (Xcode): The package product 'my-plugin' requires minimum platform version 15.0 for the iOS platform, but this target supports 12.0
...
```

To fix this, the user will need to open the iOS app in Xcode and raise its Minimum Deployment to 15.0.

> ... there is a commit for an [8.4.0 release](https://github.com/googlemaps/ios-maps-sdk/pull/7) that [supports iOS 14](https://github.com/googlemaps/ios-maps-sdk/blob/4b24150b5a48a8624808b23f36a8e2804df333e4/Package.swift#L22). Why can't we include that version in our SPM dependency range, as we do with CocoaPods?
> 
> (Is this related to [googlemaps/ios-maps-sdk#10 (comment)](https://github.com/googlemaps/ios-maps-sdk/issues/10#issuecomment-2096890239) ? There was never any clear resolution in that issue as to whether the 9.0 release did in fact fix the problem.)

Yup, [googlemaps/ios-maps-sdk#10 (comment)](https://github.com/googlemaps/ios-maps-sdk/issues/10#issuecomment-2096890239) is the blocker - CI builds will flake if SwiftPM is used to build Google Maps SDK versions `[8.4.0, 9.2.0)`. Prior to version 9.2.0, their `Package.swift` had targets with incorrect dependencies - this causes build errors if Xcode builds the targets in an unexpected order ([detailed explanation](https://github.com/googlemaps/ios-maps-sdk/issues/10#issuecomment-2096890239)). This issue was fixed in 9.2.0, which flattened their `Package.swift`'s targets.

**Comment by stuartmorgan** [comment]
That message is far better than the situation we had with CocoaPods, so while a wrapper message with instructions would be nice, I don't think we need to block on having it to proceed with dropping iOS 14 for SPM builds with `google_maps_flutter`.

**Comment by loic-sharma** [comment]
Thanks for confirming! @jokerttu We're good to drop iOS 14 support if using SPM!

**Comment by stuartmorgan** [comment]
> I don't think a broad dependency range of `[8.4.0, 10.0.0)` would allow us to lower the iOS requirement. SPM strictly enforces that a package's platform requirements are equal to or higher than its dependencies. Since GoogleMaps 9.0.0+ requires iOS 15, google_maps_flutter_ios must also require iOS 15 or higher, and the app must have a minimum deployment of iOS 15.

Wait, really? I know it's moot in this particular instance, that CocoaPods behavior is something we've benefitted from in `google_maps_flutter` for quite a while. If SPM min deployment ranges are enforced based on the *highest* version in the range, rather than the *lowest* version like CocoaPods did, we are going to be stuck in a situation where we can never allow developers to use new versions of the SDK without forcing everyone to drop iOS versions. Or, I guess, do a breaking change to the plugin every year :(

**Comment by yasargil** [comment]
We received following Info from google Maps:

> Hello Google Maps Platform customer,
> To streamline our development and release processes, and align with the evolving iOS ecosystem, we will be phasing out support for CocoaPods as an installation method for our Google Maps Platform iOS SDKs.

> We will support releasing new Google Maps Platform iOS SDKs versions (for example: version 10.x.x) through CocoaPods until Q2 2026. Starting with version 11.0 of all Google Maps Platform iOS SDKs, new versions will not be released on CocoaPods.
> Potential impact:
> If you continue to use CocoaPods after Q2 2026, you will not be able to access the latest Google Maps Platform iOS SDKs versions, including new features, performance improvements, and critical bug fixes.

**Comment by stuartmorgan-g** [comment]
Yes, we're aware of that timeline; we were already planning to have a solution here in advance of 2026Q2, so it shouldn't affect anything.

**Comment by wantroba** [comment]
@stuartmorgan-g Is the version with this adjustment already live?
